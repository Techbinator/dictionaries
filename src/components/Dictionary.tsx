import React from "react";
import {
  IDictionary,
  ITranslations,
  TaddDictionary,
  TremoveDictionary,
  TremoveTranslation,
  TeditTranslation
} from "../store/Dictionaries.type";
import NewTranslation from "./NewTranslation";
import Translation from "./Translation";

interface IDictionaryProps {
  dictionary: IDictionary;
  translations: ITranslations;
  addTranslation: TaddDictionary;
  removeDictionary: TremoveDictionary;
  removeTranslation: TremoveTranslation;
  editTranslation: TeditTranslation;
}

export default ({
  dictionary,
  addTranslation,
  removeDictionary,
  removeTranslation,
  translations,
  editTranslation
}: IDictionaryProps) => (
  <tr>
    <td className="align-middle" style={{ width: "10%" }}>
      {dictionary.name}
    </td>
    <td style={{ width: "80%" }} className="no-padding">
      <table>
        <thead>
          <tr>
            <th>Domain</th>
            <th>Range</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          <NewTranslation
            dictionaryUUID={dictionary.uuid}
            addTranslation={addTranslation}
            translations={translations}
            translationsUUID={dictionary.translationsUUID}
          />
          {dictionary.translationsUUID.map(uuid => (
            <Translation
              key={uuid}
              translationUUID={uuid}
              dictionaryUUID={dictionary.uuid}
              translations={translations}
              translationsUUID={dictionary.translationsUUID}
              removeTranslation={removeTranslation}
              editTranslation={editTranslation}
            />
          ))}
        </tbody>
      </table>
    </td>
    <td className="align-middle" style={{ width: "10%" }}>
      <button
        className="warning"
        onClick={() => removeDictionary(dictionary.uuid)}
      >
        Delete
      </button>
    </td>
  </tr>
);
