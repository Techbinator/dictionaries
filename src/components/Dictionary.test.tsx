import React from "react";
import Dictionary from "./Dictionary";
import { shallow } from "enzyme";
import contextMock from "../helpers/__mocks__/context.mock";

describe("Header component", () => {
  const wrapper = shallow(
    <Dictionary
      dictionary={
        contextMock.dictionaries["619ce930-4116-11e9-9306-ef359c18301d"]
      }
      addTranslation={contextMock.addTranslation}
      removeDictionary={contextMock.removeDictionary}
      removeTranslation={contextMock.removeTranslation}
      translations={contextMock.translations}
      editTranslation={contextMock.editTranslation}
    />
  );
  it("should render snapshot with default values", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should remove dictionary on click", () => {
    wrapper.find(".warning").simulate("click");
    expect(contextMock.removeDictionary).toBeCalledWith(
      "619ce930-4116-11e9-9306-ef359c18301d"
    );
  });
});
