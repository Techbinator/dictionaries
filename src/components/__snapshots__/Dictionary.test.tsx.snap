// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render snapshot with default values 1`] = `
<tr>
  <td
    className="align-middle"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    dictionary2
  </td>
  <td
    className="no-padding"
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <table>
      <thead>
        <tr>
          <th>
            Domain
          </th>
          <th>
            Range
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <NewTranslation
          addTranslation={[MockFunction]}
          dictionaryUUID="619ce930-4116-11e9-9306-ef359c18301d"
          translations={
            Object {
              "55a62ab0-4116-11e9-9306-ef359c18301d": Object {
                "domain": "Crimson",
                "range": "newText",
                "uuid": "55a62ab0-4116-11e9-9306-ef359c18301d",
              },
              "58f6b540-4116-11e9-9306-ef359c18301d": Object {
                "domain": "a",
                "range": "b",
                "uuid": "58f6b540-4116-11e9-9306-ef359c18301d",
              },
              "6e270e10-4116-11e9-9306-ef359c18301d": Object {
                "domain": "Turquoise",
                "range": "newRange",
                "uuid": "6e270e10-4116-11e9-9306-ef359c18301d",
              },
            }
          }
          translationsUUID={
            Array [
              "6e270e10-4116-11e9-9306-ef359c18301d",
            ]
          }
        />
        <Translation
          dictionaryUUID="619ce930-4116-11e9-9306-ef359c18301d"
          editTranslation={[MockFunction]}
          key="6e270e10-4116-11e9-9306-ef359c18301d"
          removeTranslation={[MockFunction]}
          translationUUID="6e270e10-4116-11e9-9306-ef359c18301d"
          translations={
            Object {
              "55a62ab0-4116-11e9-9306-ef359c18301d": Object {
                "domain": "Crimson",
                "range": "newText",
                "uuid": "55a62ab0-4116-11e9-9306-ef359c18301d",
              },
              "58f6b540-4116-11e9-9306-ef359c18301d": Object {
                "domain": "a",
                "range": "b",
                "uuid": "58f6b540-4116-11e9-9306-ef359c18301d",
              },
              "6e270e10-4116-11e9-9306-ef359c18301d": Object {
                "domain": "Turquoise",
                "range": "newRange",
                "uuid": "6e270e10-4116-11e9-9306-ef359c18301d",
              },
            }
          }
          translationsUUID={
            Array [
              "6e270e10-4116-11e9-9306-ef359c18301d",
            ]
          }
        />
      </tbody>
    </table>
  </td>
  <td
    className="align-middle"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    <button
      className="warning"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;
